### PR TITLE
Add proxy support for fetch_deps.

### PR DESCRIPTION
**Pull request by ryanvanoss**
http://yuilibrary.com/projects/yeti/ticket/137

### CODE REVIEW

**Comment by aversini** [comment]
+1

**Comment by reid** [comment]
I have waited to merge this in for two reasons:
1. It didn't pass the Travis build when master is passing
2. There isn't an easy way to verify this works

I shouldn't have waited on 2, since I trust you folks tested this, but 1 is weird. I'm not sure why Travis errors out on this branch while master builds OK.

I'll restart this PR's Travis build a few more times before trying to debug it.

**Comment by ryanvanoss** [comment]
Thanks Reid.

Re 2, I had the same reservation. I thought about adding a unit test, but since the issue is so closely tied to having an environment behind a proxy, it didn't seem like I could accurately test it with a vows test case. I can verify that it works for me, and my fellow team members, behind our proxy... My thoughts are that as long as it doesn't break any existing functionality, we're good to go.

As for 1, I saw that, but I didn't see an obvious reason where my change could have caused it. Let me know if I can help debug.

**Comment by reid** [comment]
Thanks @ryanvanoss! I'll take your word on 2.

I noticed the restarted Travis build is still errored. I have a local Travis-like VM that I keep around for debugging. If you'd like to help debug, you can [setup your own local Travis VM](http://reidburke.com/2013/01/28/debugging-travis-builds/) and figure out what's going on. Otherwise, I'll try getting this to fail on that VM on Wednesday (tomorrow).

I'd like to get this in Yeti's next release later this week.

**Comment by reid** [comment]
Hey @ryanvanoss, I found the problem with this patch.

All of the dep/dev files are 0-length after a clean `npm install && npm test`:

```
total 0
-rw-r--r--  1 reid  staff  0 Apr 11 15:27 dojo-doh-runner.js
-rw-r--r--  1 reid  staff  0 Apr 11 15:27 dojo.js
-rw-r--r--  1 reid  staff  0 Apr 11 15:27 expect.js
-rw-r--r--  1 reid  staff  0 Apr 11 15:27 jasmine-html.js
-rw-r--r--  1 reid  staff  0 Apr 11 15:27 jasmine.css
-rw-r--r--  1 reid  staff  0 Apr 11 15:27 jasmine.js
-rw-r--r--  1 reid  staff  0 Apr 11 15:27 mocha.css
-rw-r--r--  1 reid  staff  0 Apr 11 15:27 mocha.js
-rw-r--r--  1 reid  staff  0 Apr 11 15:26 qunit.css
-rw-r--r--  1 reid  staff  0 Apr 11 15:26 qunit.js
-rw-r--r--  1 reid  staff  0 Apr 11 15:27 yui-test.js
```

That causes the same Travis failure locally. Please fix and I'll re-review!

File: scripts/fetch_deps.js [hunk 1]
```diff
@@ -100,24 +102,19 @@ function saveURLToDep(sourceURL, filename, cb) {
...
-        var data = "";
-
-        res.setEncoding("utf8");
-
-        res.on("data", function (chunk) {
-            data += chunk;
-        });
-
-        res.on("end", function () {
             fs.writeFile(filename, data, "utf8", done);
```
Comment (reid) [inline scripts/fetch_deps.js]: Shouldn't this be `body`, not `data`?

File: scripts/fetch_deps.js [hunk 2]
```diff
@@ -100,24 +102,19 @@ function saveURLToDep(sourceURL, filename, cb) {
 
     log("Saving", sourceURL, "as", filename);
 
-    protocol.get(url.parse(sourceURL), function onResponse(res) {
+    request({
+        proxy: proxy,
+        url: sourceURL
+    }, function(error, res, body) {
+        var data = "";
```
Comment (reid) [inline scripts/fetch_deps.js]: This should probably be omitted in favor of just using `body`.

**Comment by ryanvanoss** [comment]
Thanks @reid. I'll take a look at this and get back to you. FYI, I'm guessing that I won't get a chance to dig into it until Monday at the earliest. Do you have a plan for the release following the one you're planning for this week?

**Comment by reid** [comment]
Yeti should release every week. A week ago today, I had personal stuff come up that threw off the release schedule. I plan to release today and again a week from today.

**Comment by ryanvanoss** [comment]
Hey @reid, the latest update fixes the Travis error and addresses your comments. Please review when you get a chance.

Thanks!

**Comment by reid** [comment]
I'll get to this with a lot of other Yeti changes on Tuesday. Thank you!

**Pull request merged**